### PR TITLE
Feat: Improve CI/CD

### PR DESCRIPTION
**Pull request by Meldiron**
- Remove Travis, introduce GitHub actions
- Remove psalm, introduce PHP Stan
- Add Linter
- Fix linter issues
- Fix code analysis (PHP Stan) issues
- Fix bug in `reclaim` method
- Improve tests to test for reclaim bug

### CODE REVIEW

File: src/Pools/Pool.php [hunk 1]
```diff
@@ -32,19 +32,19 @@ class Pool
     protected int $reconnectSleep = 1; // seconds
 
     /**
-     * @var array
+     * @var array<int, true>|array<Connection>
```
Comment (christyjacob4) [inline src/Pools/Pool.php:35]: When does it have two different structures for the array ?

Also does the IDE even benefit from `array<int, true>` ?? Is the IDE able to use the additional information `<int, true>`

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Left some comments

File: src/Pools/Pool.php (hunk 1 above)
Comment (Meldiron) [reply src/Pools/Pool.php:35]: Its there so PHP Stan (cod eanalyser` does not complain.

Notice in constructor we fill it with `true` values. Thats the first part.
But then when you try to `pop`, it is getting set to Connection. Thats the second part.

Comment (Meldiron) [reply src/Pools/Pool.php:35]: Changed to `@var array<Connection|true>`, that explains this scenario better.

Comment (christyjacob4) [reply src/Pools/Pool.php]: Alright 👍

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**